Fix test use date dynamically

lomamech opened the pull request:
- Use of dynamic dates
- Skip csv file import tests
- Modification of the values of the test results

lomamech committed:
Test: Use Date Dynamically in Intergration test

lomamech committed:
test: update plage date for purchase integration

lomamech committed:
Test: Fix and solve test inventory adjustement

jniles commented on server/controllers/stock/core.js:
```diff
@@ -755,9 +754,8 @@ async function getInventoryQuantityAndConsumption(params, monthAverageConsumptio
-    const y = clone.getFullYear();
-    startDate = new Date(`${y}-${m}-01`);
+
+    startDate = new Date(clone);
```
Excellent  :+1:

jmcameron reviewed (commented):
FYI, I had this failure on my pc (MySQL 8, latest node, Bhima_test):
```
  1) (/purchases) Purchases
       GET /inventory/metadata/:uuid downloads
    Checking the calculation of the order interval for inventory: Prednisolone, 0,5%, Solution, Flacon, Unité:

      AssertionError: expected 3.78 to equal 3.77
      + expected - actual

      -3.78
      +3.77

      at /home/jmcameron/Bhima/test/integration/purchase.js:253:50
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

This was odd -- off by one cent.  Most errors are larger.

Also, the End-to-End tests produced an error:
```
  1) Update Employees
       should update data for employee:
     NoSuchElementError: No element found using locator: By(css selector, [data-row="EM.TE.1"])
```

I'm not sure how to treat these errors on my local system.  Since they passed via Travis CI, I'm not sure how important my errors are.

jniles commented on test/end-to-end/purchases/purchases.spec.js:
```diff
@@ -19,11 +20,12 @@ describe('Purchase Orders', () => {
 
   it('supports single item purchase orders', async () => {
     const page = new PurchaseOrderPage();
+    const datePurchase = moment(new Date(), 'YYYY-MM-DD').subtract(1712, 'days');
```
By the way, it's helpful to make moment into a date by calling `.toDate()` on the moment object.  Like this:
```js
const datePurchase = moment(new Date(), 'YYYY-MM-DD').subtract(1712, 'days').toDate();
```

jniles commented on test/end-to-end/stock/stock.spec.js:
```diff
@@ -17,7 +17,7 @@ describe('Stock end-to-end test', () => {
   describe('Stock Entry Test', StockEntryTests);
   describe('Stock Exit Test', StockExiTests);
-  describe('Stock Import', StockImportTests);
+  describe.skip('Stock Import', StockImportTests);
```
TODO :)

jniles commented on test/end-to-end/stock/stock.z1.inventory-adjustment.js:
```diff
@@ -25,16 +25,16 @@ function StockInventoryAdjustmentTests() {
 
-    // set the VITAMINE-A to 23
-    await page.setQuantity(3, 5, 23);
+    // set the Vitamines B1+B6+B12 to 23
```
Nice 👍

jniles commented on test/integration/purchase.js:
```diff
@@ -9,11 +9,29 @@ const SearchTests = require('./purchase.search.js');
  * This test suite implements full CRUD on the /purchases HTTP API endpoint.
  */
-describe('(/purchases) Purchases', () => {
+describe.only('(/purchases) Purchases', () => {
```
We need to remove this ".only()" call.

jniles reviewed (commented):
Changes look good.  Can you remove the `.only()` call so we can be sure the tests pass?

lomamech committed:
Improvement: delete .only in intergation test for purchase

lomamech committed:
test(Improvement)
- calling .toDate in moment objetc

lomamech committed:
Test(Use Date Dynamically)
- Use of dynamic dates
- Skip csv file import tests
- Modification of the values of the test results²

lomamech committed:
test: Reset value for test
